teps/0147-tekton-artifacts-phase1.md

```diff
@@ -0,0 +1,285 @@
+  - [User Experience](#user-experience)
+  - [Performance](#performance)
+  - [Risks and Mitigations](#risks-and-mitigations)
+  - [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Implementation Plan (TBD)](#implementation-plan-tbd)
+  - [Test Plan](#test-plan)
+  - [Infrastructure Needed](#infrastructure-needed)
+  - [Upgrade and Migration Strategy](#upgrade-and-migration-strategy)
+  - [Implementation Pull Requests](#implementation-pull-requests)
+  - [References](#references)
+<!-- /toc -->
+
+## Summary
+
+[TEP-0139](https://github.com/tektoncd/community/blob/main/teps/0139-trusted-artifacts.md) introduces the Artifacts API for the use case of artifacts (backed to a Persistent Volume) that are produced and consumed within the provenance. After more reflection and discussions, we think that it is only part of the use case of the artifacts API. It does not cover the use case of artifacts like `images`, `packages`, `source code` as discussed in [this feature request](https://github.com/tektoncd/pipeline/issues/6326). We want to be able to handle different artifact types using the same API. With `StepActions` now in play, that design needs to be updated to include `StepActions`. While the Artifact API design is still undergoing early design and brainstorming, one aspect that has general agreement is around reporting of the artifact provenance data to the `TaskRun` `Status`. We therefore think that it will be nice to split the design of Artifacts support into multiple phases and incrementally add features as needed. For now, we have identified two phases to completely capture the requirements and use cases captured in [TEP-0139](https://github.com/tektoncd/community/blob/main/teps/0139-trusted-artifacts.md) and [this feature request](https://github.com/tektoncd/pipeline/issues/6326). 
+
+This TEP proposes to define the Tekton Artifacts Provenance data structure that `Steps` can output (Phase 1). It also shows how this information is carried into the `TaskRun Status`. Furthermore, it also proposes how Tekton Chains could parse the `uri` and insert the provenance data into appropriate sections of the `SLSA` Provenance payload.
+
+## Motivation
+
+As defined in the [feature request](https://github.com/tektoncd/pipeline/issues/6326), the reporing of metadata of artifacts in Tekton has grown organically and is redundant and brittle. The structure of results make extremely challenging for a remote Task to produce artifacts of unknown length (e.g. a `ko Task` that produces images from source code does not know beforehand how many images will be produced). There is a need for users to be able to surface provenance output without being limited by the result spec.
+
+This works aims to overcome these limitations.
+
+### Goals
+#### Phase 1
+
+- Provide a mechanism for surfacing artifact provenance data to the `TaskRun` status.
+- Improve the current mechanism of collecting provenance by Tekton Chains that relies on type hinting.
```

Comment:
Does this mean that this TEP includes the changes required on chains side as well?

Reply:
Synced at the Data Interfaces WG meeting. Yes, this TEP includes changes required in Chains.